### PR TITLE
Select files to mix for all rules (more use of java stream)

### LINKED ISSUE #131: Select files to mix for all rules

### LINKED ISSUE #137: More use of Java 8 Stream

### PR DESCRIPTION
**Pull request by innovimax-jim**
Closes #131 Closes #137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,45 +2,55 @@
...
-      - **join #2** will join on second column of each line
-      - **join #2,** will join on second column of first line and first colum of other lines
-      - **join #2,1** will join on second column of first line and first colum of other lines (same as previous)
-      - **join #2,,3** will join on second column of first line, first colum of second line, third column of third line and first column of other lines
-  - zip-line [#sep]: will output zip of char from file1 and file2 to fileN
-  - zip-char [#sep]: will output zip of line from file1 and file2 to fileN 
-  - zip-cell [#sep]: will output zip of cell from file1 and file2 to fileN   
-  
-Mix files togethers from a zip/jar file
+  `mix-them [char|byte] [-rule] --sel <index1>,<index2>[,<index3>...] <file1> <file2>[ <file3>... <fileN>]`
```
Comment (innovimax) [inline README.md]: mix-them [char|byte] -sel <index1> <index2>[ <index3> ...] -rule <file1> <file2>[ <file3> ...]

File: README.md [hunk 2]
```diff
@@ -2,45 +2,55 @@
...
+  
+  will generate on standard out a file based on the rule and a selection of files designed by their index
+  
+  > Here are the list of rules
+  - `file #index`: will output the umpteenth file designed by his index (ignores --sel option)
```
Comment (innovimax) [inline README.md]: why "file # index" ?
&lt;index> is enough

File: README.md [hunk 3]
```diff
@@ -2,45 +2,55 @@
...
   
-    mix-them [char|byte] [-rule] --zip zipfile
-    mix-them [char|byte] [-rule] --jar jarfile
+  `mix-them [char|byte] [-rule] --zip zipfile`  
+  `mix-them [char|byte] [-rule] --jar jarfile`
   
   will generate on standard out a file based on the rule
+  
+  `mix-them [char|byte] [-rule] --sel <index1>,<index2>[,<index3>...] --zip zipfile`  
+  `mix-them [char|byte] [-rule] --sel <index1>,<index2>[,<index3>...] --jar jarfile`
```
Comment (innovimax) [inline README.md]: same as higher

File: README.md (hunk 1 above)
Comment (innovimax-jim) [reply README.md]: ok for second choice

File: README.md (hunk 2 above)
Comment (innovimax-jim) [reply README.md]: ok, it makes sens with previous review

File: README.md (hunk 3 above)
Comment (innovimax-jim) [reply README.md]: yes

File: pom.xml [hunk 4]
```diff
@@ -33,12 +33,22 @@
...
+      			<groupId>junit</groupId>
+      			<artifactId>junit</artifactId>
+      			<version>${junit.version}</version>
+      			<scope>test</scope>		
+    		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.9.4</version>
+		</dependency>
```
Comment (innovimax) [inline pom.xml]: can you add <scope>test</scope> ?

File: pom.xml [hunk 5]
```diff
@@ -33,12 +33,22 @@
...
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.9.4</version>
+		</dependency>
```
Comment (innovimax) [inline pom.xml]: same here scope: test

File: pom.xml (hunk 4 above)
Comment (innovimax-jim) [reply pom.xml]: ok

File: pom.xml (hunk 5 above)
Comment (innovimax-jim) [reply pom.xml]: ok

**Pull request merged**